### ISSUE TITLE
VariableLayoutRenderer does not work with Custom LogManagers

### ISSUE THREAD

**Issue by grenderk** [NLog/NLog#1047]
Hello

NLog Version 4.2.0

The VariableLayoutRenderer does seem to not work if you use your own custom LogManager. The bug seems to be do using the LogManager to get the configuration file in the Append method:

VariableLayoutRender.cs (line 72-81)
```
if (LogManager.Configuration.Variables != null && LogManager.Configuration.Variables.TryGetValue(Name, out layout))
{
    //todo in later stage also layout as values?
    builder.Append(layout.Render(logEvent));
}
else if (Default != null)
{
    //fallback
    builder.Append(Default);
}
```

I have tested changing the call to LogManager to using LoggingConfiguration which seems to produce working results.
```
if(LoggingConfiguration.Variables != null && LoggingConfiguration.Variables.TryGetValue(Name, out layout))
```

Thanks.

**Comment by 304NotModified** [comment]
Thanks for the code! I will test this!

**Comment by 304NotModified** [comment]
Too bad it isn't always working: 

![image](https://cloud.githubusercontent.com/assets/5808377/11192952/20bce926-8ca4-11e5-8ab8-767fb6dfab63.png)

**Comment by 304NotModified** [comment]
It should be 

```
      var loggingConfiguration = LoggingConfiguration ?? LogManager.Configuration;
                var vars = loggingConfiguration.Variables;

```

Can you give me a test case for the "custom LogManager" ? For the unit tests

**Comment by grenderk** [comment]
[MyLogManager.cs.txt](https://github.com/NLog/NLog/files/35980/MyLogManager.cs.txt)
[Program.cs.txt](https://github.com/NLog/NLog/files/35983/Program.cs.txt)
[TestApp.nlog.txt](https://github.com/NLog/NLog/files/35984/TestApp.nlog.txt)

Attached is the code I've been using to test with
[test.txt](https://github.com/NLog/NLog/files/35987/test.txt)
and that is the resulting output of the log

**Comment by 304NotModified** [comment]
Thanks! It has been added to the test cases :)

**Comment by 304NotModified** [comment]
If appveyor is also happy, this will be released with 4.2.1 (soon).

**Comment by 304NotModified** [comment]
NLog 4.2.1 has been released! https://www.nuget.org/packages/NLog/4.2.1
